pkg/oc/cli/cmd/newbuild.go

```diff
@@ -114,6 +114,7 @@ func NewCmdNewBuild(name, baseName string, f *clientcmd.Factory, in io.Reader, o
 	cmd.Flags().StringSliceVar(&config.DockerImages, "docker-image", config.DockerImages, "Name of a Docker image to use as a builder.")
 	cmd.Flags().StringSliceVar(&config.Secrets, "build-secret", config.Secrets, "Secret and destination to use as an input for the build.")
 	cmd.Flags().StringVar(&config.SourceSecret, "source-secret", "", "The name of an existing secret that should be used for cloning a private git repository.")
+	cmd.Flags().StringVar(&config.PushSecret, "push-secret", "", "The name of an existing secret that should be used for pushing to a Docker repository if --to-docker and --to are set.")
```

Comment:
>  if --to-docker and --to are set

the way you've currently implemented this, the secret will be used regardless of whether --to/--to-docker are set.  I'd just write the help as "The name of an existing secret that should be used for pushing the output image"

Reply:
thanks, updated and added tests